.github/workflows/run-tests.yml

```diff
@@ -408,7 +408,9 @@ jobs:
           path: qgis_test_report
 
   clang-tidy:
-    if: github.event_name == 'pull_request'
+    #if: github.event_name == 'pull_request'
```

Comment:
can you back out this change please?

Reply:
I can, but it was suggested by the "clang-tidy" test author to do it beause as reported in #51917 the test is broken and would then make CI red and this PR unmergeable. See https://lists.osgeo.org/pipermail/qgis-developer/2023-February/065543.html

Reply:
\cc @troopa81

Reply:
Depending on how it's configured, it might still be possible to merge PRs with failing tests. Given the amount of issues with the test suite, I suspect it is.

Reply:
I've created a spinoff of this PR as #51935 to just contain this commit. I would wait for that to be merged before updating this PR as updating this PR would have no other effect than triggering a new CI run that's known to fail ( as you can see already in the history of this PR: https://github.com/qgis/QGIS/actions/runs/4203324613/jobs/7294511888 )

Reply:
> Depending on how it's configured, it might still be possible to merge PRs with failing tests. Given the amount of issues with the test suite, I suspect it is.

Possibly, but I've heard more than once that developers don't even consider reviewing PRs until they have a green CI icon near them, and indeed this PR only got a review AFTER I pushed the commit disabling the broken test, which I did AFTER asking for help on the mailing list and receiving the suggestion to do the disabling here in order to "get the PR merged".

Reply:
Given the observation you made about wanting to keep the clang-tidy test I've removed that commit from this PR. Can I get your approval on it now ? The CI run will fail, and I'm not even sure why github is running it again (maybe because master branch changed from the last time the code was checked)